cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/sparksql/SparkCellIterator.java

```diff
@@ -339,21 +337,28 @@ private void maybeRebuildPartition()
         }
 
         // Or new partition, so deserialize partition keys and update 'values' array
-        ByteBuffer partitionKey = rid.getPartitionKey();
-        if (numPartitionKeys == 1)
+        readPartitionKey(rid.getPartitionKey(), cqlTable, this.values, stats);
+    }
+
+    public static void readPartitionKey(final ByteBuffer partitionKey,
+                                        final CqlTable table,
+                                        final Object[] values,
+                                        final Stats stats)
+    {
+        if (table.numPartitionKeys() == 1)
         {
             // Not a composite partition key
-            CqlField field = cqlTable.partitionKeys().get(0);
-            values[field.position()] = deserialize(field, partitionKey);
+            final CqlField field = table.partitionKeys().get(0);
```

Comment:
nit: avoid using the unnecessary `final`. The project use the same code style as in https://cassandra.apache.org/_/development/code_style.html

Reply:
I don't see this check style in https://github.com/apache/cassandra-analytics/blob/trunk/config/checkstyle/checkstyle.xml